app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/interceptors/CustomAccessTokenInterceptor.kt

```diff
@@ -0,0 +1,64 @@
+package com.kks.nimblesurveyjetpackcompose.util.interceptors
+
+import com.kks.nimblesurveyjetpackcompose.repo.token.TokenRepoImpl
+import com.kks.nimblesurveyjetpackcompose.util.PREF_ACCESS_TOKEN
+import com.kks.nimblesurveyjetpackcompose.util.PREF_REFRESH_TOKEN
+import com.kks.nimblesurveyjetpackcompose.util.PreferenceManager
+import dagger.Lazy
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import java.net.HttpURLConnection
+
+class CustomAccessTokenInterceptor(
+    private val preferenceManager: PreferenceManager,
+    private val tokenRepo: Lazy<TokenRepoImpl>
```

Comment:
The dependency should be an abstraction.
```suggestion
    private val tokenRepo: Lazy<TokenRepo>
```